File: CS210FZ/lab/ReverseInteger.java
import java.util.Scanner;

public class ReverseInteger {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        System.out.println("Please input string of a signed 32-bit integer: ");
        while(true) {
            String s = in.nextLine();
            if (s.equals("q"))break;
            if (!s.matches("-{0,1}[0-9]*")) {
                System.out.println("Please input a correct integer number: ");
                return;
            }
            boolean check = checkInteger(s);
            if (check==true) {
                System.out.println(reverseInt(s));
            }else {
                System.out.println(0);
            }
            System.out.println("Please input another string of a signed 32-bit integer: ");
        }
        in.close();
    }
    public static boolean checkInteger(String num){
        if (Long.parseLong(num)<=Integer.MAX_VALUE&&Long.parseLong(num)>=Integer.MIN_VALUE){
            return true;
        }
        return false;
    }
    public static String reverseInt(String num){
        String reverseNum = "";
        String temp = "";
        if ((num.substring(0,1)).equals("-")){
            reverseNum = num.substring(0,1);
            num = num.substring(1);
        }
        int length = num.length();
        for(int i = 0; i < length; i++){
            reverseNum += num.substring(num.length()-1);
            num = num.substring(0,num.length()-1);
        }
        return reverseNum;
    }
}

File: lab/reverseInteger.py
def reverse():
    """Input the string of 32-bit number."""
    num = str(input("Input integer number: "))
    check = check_integer(num)
    if check is True:
        print(reverse_int(num))
    else:
        print(0)


def check_integer(digit):
    """Check the number whether is 32-bit number."""
    if -pow(2, 31) <= int(digit) <= (pow(2, 31) - 1):
        return True
    else:
        return False


def reverse_int(digit):
    """Reverse the string of 32-bit number."""
    num = 0
    plus_minus = ""
    reverse_num = ""
    # while True:
    #    if int(digit) < 0:
    #        num = -int(digit)
    #        digit = str(num)
    #        plus_minus = "-"
    #    reverse_num = plus_minus + digit[-1:-1]
    #    return reverse_num
    # Another way of reverse number
    if int(digit) == 0:
        return 0
    if int(digit) < 0:
        x = -int(digit)
        num = x
        plus_minus = "-"
        for i in str(num):
            reverse_num = i + reverse_num
    if int(digit) > 0:
        for i in digit:
            reverse_num = i + reverse_num
    reverse_num = plus_minus + reverse_num
    return reverse_num


reverse()
